src/scripts/components/navigation.js

```diff
@@ -27,8 +29,40 @@ export default React.createClass({
 	toggleDropdown() {
 		this.setState({ dropdown: !this.state.dropdown });
 	},
+	toggleInfoView() {
+		this.setState({ infoactive: !this.state.infoactive });
+	},
 
 	render: function() {
+
+		let infoDialog = null;
+		let activeclick = null;
+		let infoicon = null;
+
+		if(!this.state.infoactive){
+			infoicon = 'info';
+			infoDialog = null;
+			activeclick = this.toggleDropdown;
+		}else {
+			infoicon = 'times';
+			infoDialog = <InfoView onDismiss = { this.toggleInfoView} />;
+			activeclick = () => {};
+		}
+
+		let ibuttonClass = React.addons.classSet({
```

Comment:
Use `camelCase`. These variable names could be a bit more clear, what does the `i` stand for?